src/software/ai/navigator/path_planner/theta_star_path_planner.cpp

```diff
@@ -34,133 +33,131 @@ bool ThetaStarPathPlanner::isUnBlocked(int row, int col)
         }
 
         // We use the opposite convention to indicate blocked or not
-        unblocked_grid[cell] = !blocked;
+        unblocked_grid[test_coord] = !blocked;
     }
 
-    return unblocked_grid[cell];
+    return unblocked_grid[test_coord];
 }
 
-bool ThetaStarPathPlanner::isDestination(int row, int col, CellCoordinate dest)
+bool ThetaStarPathPlanner::isDestination(Coordinate test_coord, Coordinate dest)
 {
-    return (row == dest.first && col == dest.second);
+    return (test_coord.row() == dest.row() && test_coord.col() == dest.col());
 }
 
-double ThetaStarPathPlanner::calculateHValue(int row, int col, CellCoordinate dest)
+double ThetaStarPathPlanner::calculateHValue(Coordinate test_coord, Coordinate dest)
 {
     // Return using the distance formula
-    return ((double)sqrt((row - dest.first) * (row - dest.first) +
-                         (col - dest.second) * (col - dest.second)));
+    return (
+        (double)sqrt((test_coord.row() - dest.row()) * (test_coord.row() - dest.row()) +
```

Comment:
Would this be better as a static_cast<double>?

Reply:
I don't think static cast is necessary

Reply:
According to our style guide we don't use c-style casts, we should always `static_cast` when possible. There's nitty-gritty reasons as to why it's safer.

Reply:
til

Reply:
done